### PR TITLE
Configuration Improvement: Create `config.toml` on Application Start

### PR DESCRIPTION
**Pull request by kobayurii**
This PR introduces a feature to automatically create a default `config.toml` file when the application starts, ensuring that configuration parameters are loaded from environment variables. If a `config.toml` is not found, the application will generate one using the environment variables.

**Details:**
- Adds functionality to create a default `config.toml` file if it does not already exist.
- The generated `config.toml` will include parameters from the environment variables.
- Any environment variables not set will be assigned default values.

**Purpose:**
This improvement aims to simplify the configuration process and provide better visibility into the parameters being used by the application. By generating a default `config.toml`, users can easily adjust configurations without needing to manually create this file.

### CODE REVIEW

File: configuration/src/default_env_configs.rs
Comment (khorolets) [inline configuration/src/default_env_configs.rs:1]: How about putting the comments into this file to give people instructions about the values, their purpose, format and default values if not present. I believe this will help operators to figure out it quicker than looking into each line separately in the docs (doesn't exist yet) or in the repo example file.

File: configuration/src/lib.rs
Comment (khorolets) [inline configuration/src/lib.rs:138]: as a operator i would appreciate the additional info log about the successful creation of the config as a follow-up to this warning, what do you think?

**Review by khorolets — COMMENTED** [review]
Overall looks great. I left few comments where this PR might be improved. Appreciate your contribution ❤️

**Comment by kobayurii** [comment]
@race-of-sloths please, include my PR in the Race

**Comment by race-of-sloths** [comment]
@kobayurii Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Weekly streak is on the road, smart strategy! Secure your streak with another PR!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&pr=near/read-rpc/363&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/kobayurii?type=bot&pr=near/read-rpc/363&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/kobayurii)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/kobayurii)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by khorolets — CHANGES_REQUESTED** [review]
Looks great, can you please update the CHANGELOG with this change and we're ready to merge

**Review by khorolets — APPROVED** [review]
Sorry I missed that you've made the record in the CHANGELOG. Perfect!

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@khorolets, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 2 🦥

**Pull request merged**
